assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,338 @@
+        default=1,
+        help="The maximum number of parallel scoring_script runs per instance",
+    )
+    parser.add_argument(
+        "--error_threshold",
+        type=int,
+        default=-1,
+        help="The number of file failures that should be ignored",
+    )
+    parser.add_argument(
+        "--max_retries",
+        type=int,
+        default=3,
+        help="The maximum number of retries for a failed or timed-out mini batch",
+    )
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=500,  # 500sec
+        help="The timeout in seconds for scoring a single mini batch.",
+    )
+    parser.add_argument(
+        "--logging_level",
+        type=str,
+        default="info",
+        help="The log verbosity level",
+    )
+    parser.add_argument(
+        "--mini_batch_size",
+        type=int,
+        default=10,
+        help="The number of files the code_configuration.scoring_script can process in one run() call",
+    )
+    parser.add_argument(
+        "--instance_count",
+        type=int,
+        help="The number of nodes to use for each batch scoring job",
+        default=1,
+        choices=range(1, MAX_INSTANCE_COUNT),
+    )
+    parser.add_argument(
+        "--model_deployment_details",
+        type=str,
+        help="Json file to which deployment details will be written",
+    )
+    # parse args
+    args = parser.parse_args()
+    logger.info(f"Args received {args}")
+    print("args received ", args)
+
+    return args
+
+
+def create_endpoint_and_deployment(ml_client, model_id, endpoint_name, deployment_name, args):
+    """Create endpoint and deployment and return details."""
+    if args.compute_name:
+        compute_name = args.compute_name
+        try:
+            compute_cluster = ml_client.compute.get(compute_name)
+        except Exception:
+            compute_cluster = AmlCompute(
+                name=compute_name,
+                size=args.size,
+                min_instances=args.min_instances,
+                max_instances=args.max_instances,
+                idle_time_before_scale_down=args.idle_time_before_scale_down,
+            )
+            try:
+                print(f"Creating compute cluster {compute_name}")
+                ml_client.begin_create_or_update(compute_cluster).wait()
+            except Exception as e:
+                raise AzureMLException._with_error(
+                    AzureMLError.create(ComputeCreationError, exception=e)
+                )
+    else:
+        compute_name = "cpu-cluster"
+        compute_cluster = AmlCompute(
+            name=compute_name,
+            size=args.size,
+            min_instances=args.min_instances,
+            max_instances=args.max_instances,
+            idle_time_before_scale_down=args.idle_time_before_scale_down,
+        )
+        try:
+            print(f"Creating compute cluster {compute_name}")
+            ml_client.begin_create_or_update(compute_cluster).wait()
```

Comment:
Could you please add logs for successfull creation of computes.

Reply:
Added